File: frontend/types/donation.ts
export type Donation = {
    id: number;
    donorId: number;
    donorType: string;
    resourceType: string;
    quantity: number;
    description: string;
    status: string;
    createdAt: string;
};

export type DonationCreateRequest = {
    resourceName: string;
    resourceType: string;
    donorType: string;
    quantity: number;
};

File: backend/ms-donation/src/main/java/com/donaton/donation/entity/Donation.java
package com.donaton.donation.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "donations")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Donation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long donorId;

    @Enumerated(EnumType.STRING)
    private DonorType donorType;

    @Enumerated(EnumType.STRING)
    private ResourceType resourceType;

    private Integer quantity;

    private String description;

    @Enumerated(EnumType.STRING)
    private DonationStatus status;

    private LocalDateTime createdAt;
}
